src/agents/combined_context_graph.py

```diff
@@ -58,10 +27,12 @@ def guardrails_node(state: AgentState) -> AgentState:
 
 workflow.add_node("guardrails", guardrails_node)
 workflow.add_node("lookup_agent", qv_kg_lookup_agent_node)
+workflow.add_node("intent_agent", intent_agent_node)
 
 workflow.add_edge(START, "guardrails")
 workflow.add_conditional_edges("guardrails", lambda x: x["next"],
-                               {"lookup_agent": "lookup_agent", "FINISH": END})
+                               {"continue": "intent_agent", "FINISH": END})
```

Comment:
Guardrail node will either return FINISH or continue , and that will be mapped to the next agent name using this mapping dict.